webpack.config.js

```diff
@@ -0,0 +1,64 @@
+const webpack = require('webpack');
+const path = require('path');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const HtmlMinimizerPlugin = require("html-minimizer-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+
+
+module.exports = {
+    entry: './portal/index.js',
+    output: {
+        filename: '[name].js',
+        path: path.resolve(__dirname, 'pkg', 'portal', 'assets'),
+    },
+    devtool: false,
+    plugins: [
+        new MiniCssExtractPlugin(),
+        new webpack.ProvidePlugin({
+            $: 'jquery',
+            jQuery: 'jquery',
+        }),
+        new webpack.SourceMapDevToolPlugin({
+            filename: '[name].js.map',
+            exclude: ['vendors.js'],
+        }),
+        new CopyPlugin({
+            patterns: [
+                {
+                    to: path.resolve(__dirname, 'pkg', 'portal', 'assets'),
+                    context: path.resolve(__dirname, 'portal'),
+                    from: "./*.html",
+                },
+            ],
+        }),
+    ],
+    module: {
+        rules: [
+            {
+                test: /\.css$/i,
+                use: [MiniCssExtractPlugin.loader, "css-loader"],
+            },
+            {
+                test: /\.html$/i,
+                type: "asset/resource",
+            },
+        ],
+    },
+    optimization: {
+        minimize: true,
+        minimizer: [
+            `...`,
+            new HtmlMinimizerPlugin(),
```

Comment:
I would probably get rid of `HtmlMinimizerPlugin`: not a lot of benefit, but can potentially lead to issues which will be tricky to understand. This is because our html file is a Go tempalte, not a plain HTML. It probably works fine with the current content of the template, but I don't know how it is going to act with other golang template syntax.